### DISCUSSION TITLE
Just getting started...

### DISCUSSION THREAD

**Discussion by dabeer** [mathoudebine/turing-smart-screen-python#346] [Themes]
Just getting started on my own custom theme, but it seems there's some undocumented features in the sample theme that I'm hoping to get some more info about. Specifically, under CPU -> LOAD, there's "ONE", "FIVE", "FIFTEEN". Anyone know what these are? And are there others e.g. "TWENTYFIVE"?

Also, I was really hoping to use an animated background, such as a gif. Is there any hope of this feature being added in the future? hopefully near future?  Seems to me that it could be done, just by advancing the frame of the gif with every update cycle, but I haven't looked at the code yet.

**Comment by dabeer** [comment]
And now I've looked at the code and answered my own question about the LOADs:  They are 1 minute, 5 minute, and 15 minute averages, and there are no other options.

And I think an animated background using a gif could be possible. I might try to implement this myself.

**Reply by mathoudebine** [reply]
You're right about the loads: only 1/5/15min average load is available because the code uses `psutil.getloadavg()` to get these data https://psutil.readthedocs.io/en/latest/#psutil.getloadavg 

For the animated background, unfortunately it is not possible to do it with this screen for two reasons:

1. If you refresh the background during program lifecycle, it will be drawn over existing sensors data and erase the whole screen until new sensors data are displayed again, so it will not look good.
2. The background refresh is also very slow, as you can see when you start the program: it takes ~3 seconds during which no other data can be refreshed. Because of this, displaying a GIF as background is not possible.

These limitations come from the screen hardware, not from the code

**Reply by dabeer** [reply]
What if some sort of buffering system was implemented. Instead of rendering sensors directly to the lcd, render them to a buffer. Then, as fast as the screen will allow, render the contents of the buffer onto the frame from the animated background (I'm thinking gif here), then render the result onto the screen?  You'd be doing a full screen refresh every ~3 sec or so, which might be CPU intensive or heat generating, but you could get a very slow animation that way?

**Reply by dabeer** [reply]
Another issue that might be solved by a buffering system is nesting radial sensors.  I tried to set this up but whichever sensor is rendered last overwrites the previous sensor with an empty black square. Buffering could use alpha values, preventing overwriting and also allowing erasing. 

The tradeoff, of course, is that full screen refreshes would be required - and even, only full screen refreshes would be possible. Still... maybe I give it a go.
